File: docker-images/python-base/runner.py
import sys
import os

def main():
    if len(sys.argv) < 2:
        print("No path to user code provided.")
        return

    user_code_path = os.path.join(sys.argv[1], "user_code.py")
    try:
        with open(user_code_path, "r") as f:
            code = f.read()
            exec(code, {})
    except Exception as e:
        print(f"Execution error: {e}")

if __name__ == "__main__":
    main()

File: docker-images/js-base/runner.js
const fs = require('fs');
const path = require('path');

const userCodeDir = process.argv[2];
const userCodePath = path.join(userCodeDir, 'user_code.js');

fs.readFile(userCodePath, 'utf8', (err, code) => {
    if (err) {
        console.error("Error reading user code:", err);
        process.exit(1);
    }
    try {
        eval(code);
    } catch (e) {
        console.error("Execution error:", e);
        process.exit(1);
    }
});
